### PR TITLE
[MINOR] Remove all unused import

### PR DESCRIPTION
**Pull request by MyLanPangzi**
MyLanPangzi [Closes #417](https://github.com/yaooqinn/kyuubi/pull/417)    Target Issue Test Plan [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

### _Why are the changes needed?_
remove all unused import and replace deprecated class PersistentEphemeralNode with PersistentNode in ServiceDiscovery

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1201,6 +1201,12 @@
             <plugin>
                 <groupId>net.alchim31.maven</groupId>
                 <artifactId>scala-maven-plugin</artifactId>
+                <configuration>
```
Comment (yaooqinn) [inline pom.xml:1204]: can we move these to L1022

File: kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ServiceDiscovery.scala [hunk 2]
```diff
@@ -24,17 +24,16 @@ import java.util.concurrent.atomic.AtomicBoolean
 import javax.security.auth.login.Configuration
 
 import scala.collection.JavaConverters._
-import scala.util.control.NonFatal
 
 import org.apache.curator.framework.{CuratorFramework, CuratorFrameworkFactory}
-import org.apache.curator.framework.recipes.nodes.PersistentEphemeralNode
+import org.apache.curator.framework.recipes.nodes.PersistentNode
```
Comment (ulysses-you) [inline kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ServiceDiscovery.scala:29]: @MyLanPangzi This is not just remove unused import but also remove some warn, please note the change at the title and description.

**Comment by pan3793** [comment]
The CI failed because disk capacity issue, can we enlarge disk capacity or reduce cache size?

**Comment by MyLanPangzi** [comment]
updated title and descrption. the ci was wrong.
how to fix the error ? 
error message :
System.IO.IOException: No space left on device Unhandled exception. System.IO.IOException: No space left on device at System.IO.FileStream.WriteNative(ReadOnlySpan`1 source) at System.IO.FileStream.FlushWriteBuffer() at System.IO.FileStream.Flush(Boolean flushToDisk) at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder) at System.Diagnostics.TextWriterTraceListener.Flush() at System.Diagnostics.TraceSource.Flush() at GitHub.Runner.Common.TraceManager.Dispose(Boolean disposing) at GitHub.Runner.Common.TraceManager.Dispose() at GitHub.Runner.Common.HostContext.Dispose(Boolean disposing) at GitHub.Runner.Common.HostContext.Dispose() at GitHub.Runner.Worker.Program.Main(String[] args) at System.IO.FileStream.WriteNative(ReadOnlySpan`1 source) at System.IO.FileStream.FlushWriteBuffer() at System.IO.FileStream.Flush(Boolean flushToDisk) at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder) at System.Diagnostics.TextWriterTraceListener.Flush() at GitHub.Runner.Common.HostTraceListener.WriteHeader(String source, TraceEventType eventType, Int32 id) at GitHub.Runner.Common.HostTraceListener.TraceEvent(TraceEventCache eventCache, String source, TraceEventType eventType, Int32 id, String message) at System.Diagnostics.TraceSource.TraceEvent(TraceEventType eventType, Int32 id, String message) at GitHub.Runner.Worker.Worker.RunAsync(String pipeIn, String pipeOut) at GitHub.Runner.Worker.Program.MainAsync(IHostContext context, String[] args) System.IO.IOException: No space left on device at System.IO.FileStream.WriteNative(ReadOnlySpan`1 source) at System.IO.FileStream.FlushWriteBuffer() at System.IO.FileStream.Flush(Boolean flushToDisk) at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder) at System.Diagnostics.TextWriterTraceListener.Flush() at GitHub.Runner.Common.HostTraceListener.WriteHeader(String source, TraceEventType eventType, Int32 id) at GitHub.Runner.Common.HostTraceListener.TraceEvent(TraceEventCache eventCache, String source, TraceEventType eventType, Int32 id, String message) at System.Diagnostics.TraceSource.TraceEvent(TraceEventType eventType, Int32 id, String message) at GitHub.Runner.Common.Tracing.Error(Exception exception) at GitHub.Runner.Worker.Program.MainAsync(IHostContext context, String[] args)

File: pom.xml [hunk 3]
```diff
@@ -88,7 +88,7 @@
         <maven.plugin.scala.version>4.4.0</maven.plugin.scala.version>
         <maven.plugin.surefire.version>2.22.0</maven.plugin.surefire.version>
         <maven.plugin.scalatest.version>2.0.0</maven.plugin.scalatest.version>
-        <maven.plugin.scalatest.exclude.tags></maven.plugin.scalatest.exclude.tags>
+        <maven.plugin.scalatest.exclude.tags/>
```
Comment (pan3793) [inline pom.xml:91]: I would rather keep it as is.

File: kyuubi-main/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala [hunk 4]
```diff
@@ -111,7 +112,7 @@ trait ProcBuilder {
         var line: String = reader.readLine
         while (true) {
           if (containsIgnoreCase(line, "Exception:") &&
-              !line.contains("at ") && !line.startsWith("Caused by:")) {
+            !line.contains("at ") && !line.startsWith("Caused by:")) {
```
Comment (yaooqinn) [inline kyuubi-main/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala:115]: let's keep it

**Comment by yaooqinn** [comment]
w.r.t our instructions in  the PR description, you can debug locally first

File: kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ServiceDiscovery.scala [hunk 5]
```diff
@@ -24,17 +24,16 @@ import java.util.concurrent.atomic.AtomicBoolean
 import javax.security.auth.login.Configuration
 
 import scala.collection.JavaConverters._
-import scala.util.control.NonFatal
 
 import org.apache.curator.framework.{CuratorFramework, CuratorFrameworkFactory}
-import org.apache.curator.framework.recipes.nodes.PersistentEphemeralNode
```
Comment (yaooqinn) [inline kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ServiceDiscovery.scala:30]: Kindly ping @MyLanPangzi, any update on this PR?

If you have any problem with local debugging, please feel free to let us know.

BTW, we can make a separate PR for this and other import optimization

**Comment by yaooqinn** [comment]
You can rebase the master branch and push. The unit-tests.log where the true bug laying can be upload by CI now, and then everybody can see and help.

**Comment by pan3793** [comment]
spark 3.0 bundles curator 2.7.1 and spark 3.1 bundles curator 2.13.0, I guess the zk related changes may not work on curator 2.7.1.
@MyLanPangzi could you please separate zk related changes into another PR, and test both with curator 2.7. 1& 2.13.0?

**Comment by MyLanPangzi** [comment]
wait a moment.

**Comment by MyLanPangzi** [comment]
compile error,beause the deprecated api.
Should this configuration be removed?  <arg>-Xfatal-warnings</arg>


[INFO] Compiling 7 Scala sources to E:\github\kyuubi\kyuubi-ha\target\scala-2.12\classes ...
[ERROR] E:\github\kyuubi\kyuubi-ha\src\main\scala\org\apache\kyuubi\ha\client\ServiceDiscovery.scala:61: class PersistentEphemeralNode in package nodes is deprecated: see corresponding Javadoc for more information.
[ERROR] E:\github\kyuubi\kyuubi-ha\src\main\scala\org\apache\kyuubi\ha\client\ServiceDiscovery.scala:68: class PersistentEphemeralNode in package nodes is deprecated: see corresponding Javadoc for more information.
[ERROR] E:\github\kyuubi\kyuubi-ha\src\main\scala\org\apache\kyuubi\ha\client\ServiceDiscovery.scala:121: class PersistentEphemeralNode in package nodes is deprecated: see corresponding Javadoc for more information.
[ERROR] three errors found
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  20.294 s
[INFO] Finished at: 2021-03-15T16:13:06+08:00
[INFO] ------------------------------------------------------------------------

**Comment by MyLanPangzi** [comment]
i removed the configuration, but local test cannot pass.
my os is win10.

  Cause: java.nio.file.InvalidPathException: Illegal char <:> at index 2: /E:/github/kyuubi/kyuubi-ha/target/scala-2.12/test-classes/\kyuubi-kdc-b1a8dae9-54be-4be0-be88-ef937f6c2869
  at sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
  at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
  at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
  at sun.nio.fs.WindowsPath.parse(WindowsPath.java:94)
  at sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:255)
  at java.nio.file.Paths.get(Paths.java:84)
  at org.apache.kyuubi.Utils$.$anonfun$createDirectory$1(Utils.scala:86)
  at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:158)
  at org.apache.kyuubi.Utils$.createDirectory(Utils.scala:85)
  at org.apache.kyuubi.Utils$.createTempDir(Utils.scala:105)
  ...

**Comment by pan3793** [comment]
> Should this configuration be removed? -Xfatal-warnings

It there possible only skip specific use? It should be something like `<excludes>...` on maven plugin conf or annotation on code.

**Comment by yaooqinn** [comment]
the current tests do not support windows, seems like that you can only depend on Github actions

**Comment by yaooqinn** [comment]
> > Should this configuration be removed? -Xfatal-warnings
> 
> It there possible only skip specific use? It should be something like `<excludes>...` on maven plugin conf or annotation on code.



> > Should this configuration be removed? -Xfatal-warnings
> 
> It there possible only skip specific cases? It should be something like `<excludes>...` on maven plugin conf or annotation on code.

Try
```
<arg>-P:silencer:globalFilters=.*deprecated.*</arg>
```
```
https://mvnrepository.com/artifact/com.github.ghik/silencer-plugin_2.12.10/1.6.0
```
For `scala-maven-plugin`, you can set `compilerPlugins` in the `configuration`
```xml
         <compilerPlugins>
              <compilerPlugin>
                <groupId>com.github.ghik</groupId>
                <artifactId>silencer-plugin_${scala.version}</artifactId>
                <version>1.6.0</version>
              </compilerPlugin>
            </compilerPlugins>

```

**Comment by MyLanPangzi** [comment]
ok, it's working !
i reverted org.apache.kyuubi.ha.client.ServiceDiscovery change.

**Review by yaooqinn — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks! I will merge it.
